### DISCUSSION TITLE
parse invalid cbor and continue without exceptions

### DISCUSSION THREAD

**Discussion by jprochazk** [nlohmann/json#2291] [Q&A] [answered]
I have a hot-path in my code that every user message goes through. I want to validate every incoming message, and discard it if the message is invalid. Looking through the documentation, there doesn't seem to be an alternative to exceptions. How can I parse CBOR, without using exceptions, but also without aborting, so that the application can continue?

**Comment by oficsu** [comment]


**Reply by jprochazk** [reply]
I'm trying to parse CBOR, not JSON, and the interface isn't the same. I don't know how to provide a similar overload for `from_cbor`.

**Comment by jprochazk** [comment, marked as answer]
I believe I found a bug, or at least unexpected(?) behavior. I made an issue https://github.com/nlohmann/json/issues/2293
